website/blog/2021-11-21-2.5.0.md

```diff
@@ -0,0 +1,384 @@
+
+Support new TypeScript 4.5 features.
+
+##### [`type` Modifiers on Import Names](https://devblogs.microsoft.com/typescript/announcing-typescript-4-5/#type-on-import-names)
+
+<!-- prettier-ignore -->
+```ts
+// Input
+import { type A } from "mod";
+
+// Prettier 2.4
+SyntaxError: ',' expected. (1:15)
+
+// Prettier 2.5
+import { type A } from "mod";
+
+```
+
+##### [Private Field Presence Checks](https://devblogs.microsoft.com/typescript/announcing-typescript-4-5/#private-field-presence-checks)
+
+<!-- prettier-ignore -->
+```ts
+// Input
+class Foo {
+  #prop1;
+  method() {
+    return #prop1 in this;
+  }
+}
+
+// Prettier 2.4
+SyntaxError: Private identifiers are not allowed outside class bodies. (4:12)
+  2 |   #prop1;
+  3 |   method() {
+> 4 |     return #prop1 in this;
+    |            ^
+  5 |   }
+  6 | }
+  7 |
+
+// Prettier 2.5
+class Foo {
+  #prop1;
+  method() {
+    return #prop1 in this;
+  }
+}
+
+```
+
+##### [Import Assertions](https://devblogs.microsoft.com/typescript/announcing-typescript-4-5/#import-assertions)
+
+<!-- prettier-ignore -->
+```ts
+// Input
+import obj from "./something.json" assert { type: "json" };
+
+// Prettier 2.4
+SyntaxError: ';' expected. (1:36)
+> 1 | import obj from "./something.json" assert { type: "json" };
+    |                                    ^
+  2 |
+
+// Prettier 2.5
+import obj from "./something.json" assert { type: "json" };
+
+```
+
+##### Handle `.mts` and `.cts`
+
+Handle new file extensions `.mts` and `.cts`.
+
+### MDX
+
+#### Add support for MDX 2 comments ([#11563](https://github.com/prettier/prettier/pull/11563) by [@wooorm](https://github.com/wooorm))
+
+This adds basic support for MDX 2 comments (JavaScript-style comments) in addition to existing MDX 1 comments (HTML-style comments).
+
+Note that in MDX 2, more complex comments and whole expressions can be used inside braces.
+This currently only supports a single comment on one line so that `{/* prettier-ignore */}` can be used, but doesn’t support most of MDX 2.
+
+<!-- prettier-ignore -->
+```mdx
+Input:
+{/*A comment*/}
+
+Prettier 2.4:
+{/_A comment_/}
+
+Prettier 2.5:
+{/*A comment*/}
+```
+
+## Other Changes
+
+### JavaScript
+
+#### Fix parentheses around sequence expression as body of arrow chain ([#11593](https://github.com/prettier/prettier/pull/11593) by [@bakkot](https://github.com/bakkot))
+
+The required parentheses around sequence expressions as the body of arrow functions were dropped for chained arrows, causing incorrect code to be generated. Now they are included.
+
+<!-- prettier-ignore -->
+```jsx
+// Input
+const f = () => () => (0, 1);
+
+// Prettier 2.4
+const f = () => () => 0, 1;
+
+// Prettier 2.5
+const f = () => () => (0, 1);
+```
+
+#### Ignore errors for sloppy mode syntax ([#11750](https://github.com/prettier/prettier/pull/11750) by [@fisker](https://github.com/fisker), [#11778](https://github.com/prettier/prettier/pull/11778) by [@sosukesuzuki](https://github.com/sosukesuzuki))
+
```

Comment:
```suggestion

JavaScript’s [strict mode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode) adds several useful errors to prevent mistakes. Some of these errors are syntax errors that occur at parse time. Since Prettier’s goal is to format all syntactically valid JavaScript code regardless of whether it will actually run, we’ve opted to leave this error checking to linters, compilers, and the runtime.

```